LiteCore/Query/QueryParser.cc

```diff
@@ -1007,6 +1009,36 @@ namespace litecore {
     // Handles object (dict) property accessors, e.g. ["_.", [...], "prop"] --> fl_nested_value(..., "prop")
     void QueryParser::objectPropertyOp(slice op, Array::iterator& operands) {
         auto nOperands = operands.count();
+        
+        // Handle Meta().id - N1QL
+        // ["_.", ["meta" <db>], ".id"] - JSON
+        do {
+            if (nOperands != 2) {
+                break;
+            }
+            const Array* metaop = operands[0]->asArray();
+            if (metaop == nullptr || metaop->count() == 0 ||
+                metaop->begin().value()->asString() != "meta"_sl) {
+                break;
+            }
+            slice dbAlias;
+            if (metaop->count() > 1) {
+                const Value* second = metaop->get(1);
+                if (second->type() == kString) {
+                    dbAlias = second->asString();
+                }
+            }
+            slice meta_key = operands[1]->asString();
+            if (meta_key == nullptr) {
+                break;
+            }
+            if (meta_key[0] == '.') {
+                meta_key.moveStart(1);
+            }
+            writeMetaPropertyGetter(meta_key.asString(), dbAlias.asString(), true);
```

Comment:
There doesn't seem any reason to convert these into `std::string`...

Reply:
Right. But the second argument needs to be converted to string, according to the signature of the method. I will change the first one.